### ISSUE TITLE
Help with running from Docker image install

### ISSUE THREAD

**Issue by froggiie** [taunometsalu/ClustVis#16]
Hi, firstly just want to say thanks for creating and maintaining an excellent tool. 

I was hoping you could help me run your great app locally (the website can be a bit slow if I'm trying many variations).

I am new to Docker, but understand it to be a way of running controlled environments and fixed software versions. I haven't used it before. Unfortunately, I haven't quite got it running but hope that I'm close.

**_My process for installing_** 

**Running a Windows 10 machine, I downloaded docker for windows.  From CMD prompt I downloaded the docker image using "docker pull taunometsalu/clustvis.**"

**From within Docker, I installed ClustVis with the commands (entered verbatim in to CLI):**

sudo docker pull taunometsalu/clustvis
mkdir ~/customClustvis/
cd ~/customClustvis/
wget https://github.com/taunometsalu/ClustVis/archive/master.zip
unzip master.zip
chmod -R go+rx ~/customClustvis/
sudo docker run -d \
	--name customClustvis \
	-p <myPort>:3838 \
    -v ~/customClustvis/ClustVis-master/:/srv/shiny-server/:ro \
    taunometsalu/clustvis

**Now when I run from within docker, it says:**

[2022-04-01 00:20:33.349] [INFO] shiny-server - Shiny Server v1.5.5.872 (Node.js v6.10.3)

[2022-04-01 00:20:33.351] [INFO] shiny-server - Using config file "/etc/shiny-server/shiny-server.conf"

[2022-04-01 00:20:33.374] [WARN] shiny-server - Running as root unnecessarily is a security risk! You could be running more securely as non-root.

[2022-04-01 00:20:33.376] [INFO] shiny-server - Starting listener on 0.0.0.0:3838

Unfortunately, that's as far as I got. I tried entering http:/0.0.0.0.3838 in to chrome and IE web browser (from my windows 10, not any linux version within CMD), but don't see any shiny app. I'm not sure how my Windows 10 interacts with the docker image. Do I need to install or run something else? Is it a mistake that the IP is showing as 0.0.0.0?

Thank you again.

Kind regards,
Richard

**Comment by taunometsalu** [comment]
Hi Richard, yes, you are close. Some changes are needed:
- please use `-p 3838:3838` instead of `-p :3838` when running the container
- did you literally enter `http:/0.0.0.0.3838`? There should be double slash and a colon like `http://0.0.0.0:3838`. Or, if it still doesn't work then you can try `http://host.docker.internal:3838` or `http://127.0.0.1:3838`.

**Comment by froggiie** [comment]
Hi, thank you for getting back to me. I figured it out. The problem was that I really didn’t understand how WSL (Windows Subsytem of Linux) worked with the install commands. I now recognise that I could download the docker image, but I hadn’t really performed the subsequent steps because they weren’t working in command prompt (or powershell) natively as UNIX commands (at least not verbatim).

**_How I installed  and ran a local server on a Windows machine:_**

Install Docker for windows. 

Install Ubuntu for Windows 10 (windows store). Set up your user name and password when prompted. 

I couldn’t use it native with “docker” commands from the Ubuntu command line because it needed to be configured as WSL2.
So I converted Ubuntu on WSL1 to WSL2. I did this with the commands described here:
https://ubuntu.com/blog/ubuntu-on-wsl-2-is-generally-available

Once this was done, I enabled Ubuntu within the docker app by going to Settings>Resources>WSL integration>turn on Ubuntu

After this, from the Ubuntu command line I inputted the following:

sudo docker pull taunometsalu/clustvis
mkdir ~/customClustvis/
cd ~/customClustvis/
wget https://github.com/taunometsalu/ClustVis/archive/master.zip
unzip master.zip
chmod -R go+rx ~/customClustvis/
sudo docker run -d \
	--name customClustvis \
	-p 3838:3838 \
    -v ~/customClustvis/ClustVis-master/:/srv/shiny-server/:ro \
    taunometsalu/clustvis

Then Docker ran the customClustvis enrironment and I could access it using http://host.docker.internal:3838

Thank you again for this excellent tool and your help!
